Local multiplayer implemented

HaschkaRa opened the pull request:
# README:
Currently, thismultiplayer is mergeable with the main branch without conflicts (13.4. 4:50). However, if the main branch changes before this merge goes through, I will create a new request in case of larger merge conflicts


## Initial working multiplayer demo with host/client connection and player instancing
- Implemented basic multiplayer setup using ENetMultiplayerPeer.
- Added buttons to host or join a session on UI.
- Players are instantiated on connection with correct multiplayer authority
- Camera only activates for local (authoritative) player.
- Movement and jumping are handled locally for the authority peer.
- Players are spawned into a simple 3D environment (map).
- Confirmed player spawning / syncing works, but spawning order can still cause bugs if host/client race occurs.

## First local multiplayer implementation
- Added PlayerInfo.cs -> to save and sync player information (Name, Id)
- Added a Play button to localMenu.tscn to start game after players joiined
- implemented Play button logic
- implemented basic local multiplayer functionality in LocalMultiplayer.cs
- - implemented multiplayer server / client logic
- - implemented
- added GameManager.cs with a list to store all player Infos in a lobby
- added PlayerSpawnPoints node with 4 spawn points in a global group
- added MultiplayerSynchronizer in default_player.tscn to handle player syncing and implemented it in DefaultPlayer.cs
- handling player connection, disconnection, connection to server and failed connection
- implemented host functionalitys
- implemented join functionalitys
- syncing players information such as name and id with server / host and transmitting it to all other Players

## Enhanced player spawning and enable enemy spawning for multiplayer support
- Removed the default player from Main.tscn because in multiplayer, a player is spawned automatically for each peer.
- Added a Main.cs script to the main scene to handle player spawning.
- Set the FPS limit to 60 in Main.cs to significantly reduce GPU load.
- Removed the Joystick node from default_player.tscn; it is now instantiated individually per player (so each player only sees their own joystick).
- Fixed a bug where multiple instances of the same player were being spawned.
-Made the server/host the central authority for managing the list of active players in the lobby when the game starts.
- Moved Path2D, PathFollow2D, and SpawnTimer from the main scene to a dedicated SpawnEnemies scene in the Scenes/Enemies folder to support multiplayer functionality.
- Moved SpawnEnemies.cs to the Scripts/Enemies folder.
- Assigned each player instance to a specific SpawnEnemies instance.

Pazl27 reviewed (changes requested):
Looks good and runs on my machine. 
Maybe someone else who is more experienced with godot should look at it besides me. 
I found some small issues:
- [ ] the back buttons in the menu don't work.

I would suggest waiting for the [branch 118](https://github.com/IP-2025/waves-of-the-fallen/pull/119) this branch has the same issue, to be fixed and merged. And then merge the master into your branch. And check if everything works

HaschkaRa commented:
- Fixed enemy behavior. The issue was that I set the spawner as the player's child, but it is now the map's child.
- The back button works on my machine though.

- We also added the first multiplayer sample in the First_Prototype folder (command line script, does not run in Godot), but this has nothing to do with the game and does not affect it

manuel0804 commented:
Seems pretty good and everything works fine now. No Problems with back buttons on my machine too.

Pazl27 reviewed (changes requested):
Dunno if i would put the prototypes into the repo, they dont really matter for the game i would only put stuff that really matters to the game into the repo. They make it a little bit bloated the repo and maybe harder to find stuff inside and a bit confusing for outstanding people. If you want to keep these prototypes saved somewhere we can create a new repo insdie the org or just upload them to your own github

the rest looks good and works thanks
